packages/seacas/libraries/ioss/src/Ioss_ElementTopology.h

```diff
@@ -117,14 +122,20 @@ namespace Ioss {
     bool operator!=(const Ioss::ElementTopology &rhs) const;
     bool equal(const Ioss::ElementTopology &rhs) const;
 
+    virtual ElementPermutation *permutation() const;
+    virtual const std::string &base_topology_permutation_name() const;
```

Comment:
I guess it is virtual so can be overridden in spring and node and sphere....  I guess that makes sense. Or do we need to have a different value in the `ElementShape` enum for each of those?

Reply:
I wanted to add new values to `ElementShape` for spring, node and sphere but I wasn't sure if this would have repercussions in the rest of the code ... if it's ok to do that, then I can add them

Reply:
It should not affect anything else.  The only place that the ElementShape is currently used is in the CGNS code which maps between IOSS element types and CGNS types and those are primarily/exclusively solid elements.  If not, I can add code to make it work with spring and node and sphere...